Zadanie 1. Oblicz $\frac{2^2 + \sqrt{21}\cdot (-5)^3 - 3^2}{2^{\frac{1}{3}} + 4}$
i  $\frac{2^2 + \sqrt{21}\cdot ((-5)^3 - 3^7)}{2^{\frac{1}{3}} + 4}$.

Wskazówka 1: Pierwiastek z liczby r zwraca polecenie: r.sqrt\(\)   lub   sqrt\(r\).

Zauważysz, że jeżeli r jest liczbą wymierną, z której pierwiastek jest niewymierny, to powyższe polecenie zwróci dokładną wartość tzn. w odpowiedzi użyje znaku pierwiastka \(chociaż maksymalnie to wyrażenie uprości\). Jeżeli chcemy uzyskać wartość przybliżoną, musimy po r postawić kropkę, np

sqrt\(21.\)   lub  21.0.sqrt\(\)

Wskazówka 2: Potęgę $n^k$ wyliczymy wpisując jedno z poleceń: 

n^k lub n**k  \(preferowane\).

Wskazówka 3: Kolejność działań w jakiej Sage wykonuje obliczenia jest zgodna z ogólnie przyjętą kolejnością działań. Jeżeli chcemy np. najpierw dodać dwie liczby a ich sumę pomnożyć przez trzecią, stosujemy nawiasy półokrągłe: \(a\+b\)⋅c. 


In [1]:
(2^2 + sqrt(21.)*((-5)^3 - 3^2))/(2^(1.0/3.0)+4)

-115.983707233833

In [2]:
(2^2 + sqrt(21.)*((-5)^3 - 3^7))/(2^(1.0/3.0)+4)

-2013.51216230699

In [3]:
1/(2.1-2)

9.99999999999999

In [8]:
1/((21/10)-2)

10

In [9]:
  floor((1/10)^14)

0

In [10]:
(1/10)^14

1/100000000000000

Zadanie 2: 

21 lat temu kwota 200$ została wpłacona na rachunek oszczędnościowy, którego oprocentowanie wynosi 5% w skali roku. Ile jest teraz (czy w 21 rocznicę jego założenia) pieniędzy na rachunku?

In [17]:
200*((1+(5.0/100))^(21))

557.192518080329

Zadanie 3:

Pewnego razu w odległej galaktyce dwie planety zamieszkane przez istoty inteligentne okrążały gwiazdę. Doba na jednej z nich trwała 15h na drugiej - 25h. Podróżnik przybył do tego układu, gdy na obu planetach zegary pokazywały godzinę 0. Ile czasu minie, nim zegary na obu planetach znowu pokażą tę godzinę? (Wyznaczyć najkrótszy z takich przedziałów czasu.)

Mieszkańcy obu planet postanowili wprowadzić dodatkową (oprócz godziny i doby) jednostkę czasu. Powinna spełniać dwa warunki:

- musi być dłuższa od godziny, krótsza od doby
- zawierać całkowitą liczbę godzin
- doba powinna składać się z całkowicie wielu takich jednostek
- powinna być możliwie najdłuższą jednostką czasu spełniającą powyższe warunki.

Ile godzin powinna liczyć ta nowa jednostka?

In [2]:
lcm(15,25)

75

Odpowiedź: Minie 75 godzin.

W drugiej części zadania, liczbami spełniającymi warunki jest 2 i 3. Szukamy największej z nich:

In [6]:
max(2,3)

3

Odpowiedź: Nowa jednostka powinna liczyć 3 godziny.

Zadanie 4:

W pewnym teleturnieju jego uczestnicy muszą szybko wskazać najmniejszą i największą z podanego zbioru liczb. Pomóż im w tym zadaniu, wiedząc, że te liczby to:

$1273^{5}$, 5, π3, $e^e$, $\frac{4^4}{3^3}$,$2^{5/12}$, ∣3π−π3∣

In [11]:
max(1273**5, pi**3, (e.n())**(e.n()), (4**4/(3**3)), abs(3**pi - pi**3))

3343043427317593

In [27]:
min(1273**5, pi**3, (e.n())**(e.n()), (4**4/(3**3)), abs(3^pi - pi^3))

abs(-pi^3 + 3^pi)

In [28]:
floor(abs(3^pi - pi^3))

0

Zadanie 5:

Pewien matematyk zakodował czterocyfrowy PIN swojej karty kredytowej w następujący sposób:

- liczba dwucyfrowa, której liczba dziesiątek jest równa pierwszej liczbie PIN-u a liczba jedności - drugiej jest równa części całkowitej ilorazu największego pierwszego podzielnika 1070 przez najmniejszy pierwszy podzielnik 1070.
- ostatnie dwie cyfry uzyskamy wyliczając najmniejszą liczbę całkowitą większą od większej z liczb: $e^π$
 i $π^e$.

In [31]:
1070.prime_divisors()

[2, 5, 107]

In [32]:
min(2,5,107)

2

In [33]:
max(1,5,107)

107

In [34]:
floor(107/2)

53

Z tego wynika, że pierwsza cyfra PIN-u to 3, a druga to 5.

In [35]:
floor(min(e^pi, pi^e))

22

Odpowiedź: Szukany PIN to : 3522.

Zadanie 6:
Pewna fundacja zajmująca się rekultywacją terenów postindustrialnych otrzymała wsparcie finansowe w wysokości 73 428,10 zł. Za tę sumę ma zamiar kupić sadzonki dębów błotnych (w liczbie tak dużej, jak to tylko możliwe), a jeśli po zakupie zostana jakieś pieniądze, to wyda je na zakup posiłków dla wolontariuszy sadzących dęby. Sadzonka dębu błotnego kosztuje 320 zł. Ile dębów zostanie zasadzonych i ile pieniędzy zostanie na zakup prowiantu dla wolontariuszy?

In [37]:
floor((73428.10)/320)

229

Maksymalna liczba zakupionych sadzonek dębów wynosi 229.

In [43]:
n(((723428.10)-(229*320)), digits= 8)

650148.10

Odpowiedź: Zostanie zasadzonych 229 dębów. Na zakup prowiantu zostanie 650148.10 zł.

In [48]:
n(e**(pi*i), digits=4)

-1.000

In [51]:
(pi==7)**2

pi^2 == 49

In [54]:
n(pi^2, digits=21)

9.86960440108935861883

In [56]:
var('z y')

(z, y)

In [57]:
(z+y)+(z-y)

2*z

In [61]:
y=2

In [66]:
(y-z)+(y+z)

4

In [67]:
var('a')
simplify((a*x)/(z*x))

a/z

In [68]:
factor(x**2-a**2)

-(a + x)*(a - x)

In [69]:
expand((a-x)*(a+x))

a^2 - x^2

In [70]:
a=2
z=3
round(a/z, 8)

0.66666667

In [71]:
sin(π)

0

In [74]:
n(sin(π/3), digits=3)

0.866

In [76]:
n(exp(2), digits=3)

7.39

In [80]:
n(log(e), digits=10)

1.000000000

In [81]:
n(ln(e), digits=10)

1.000000000

In [83]:
n(log(10), digits=3)

2.30

In [84]:
n(ln(10), digits=3)

2.30

In [85]:
sqrt(ln(e^4))

2

In [89]:
n((sin(pi/4))/(cos(pi)/6), digits=3)

-4.24

In [10]:
f(x)=(2/3)*x^2

In [5]:
f.subs(x=2)

x |--> 8/3

In [7]:
f.subs({x:2})

x |--> 8/3

In [8]:
f(x=2)

8/3

In [13]:
f(2)

8/3

In [14]:
x

x

In [15]:
var('u')

u

In [18]:
g(u,x)=x^2-u

In [20]:
g(3,2)

1

In [21]:
g(x=2)

-u + 4

In [22]:
var('q r n')

(q, r, n)

In [23]:
k(q,r,n)= q^3-r/n

In [24]:
k(2,3,4)

29/4

In [26]:
var('a y t o n')

(a, y, t, o, n)

In [27]:
a=(1 + (1/n))^n

In [29]:
lim(a, n=oo)

e

In [30]:
y=(3/2)^n

In [31]:
lim(y,n=oo)

+Infinity

In [32]:
t=(-1)^n

In [34]:
lim(t,n=oo)

ind

In [35]:
o=n^928-n^516218

In [36]:
lim(o,n=oo)

-Infinity

In [37]:
h1(x)=sin(x^2+1)

In [59]:
lim(h1, x=0)

x |--> sin(1)

In [58]:
lim(h1,x=pi)

x |--> sin(pi^2 + 1)

In [63]:
h2(x)=3*x

In [56]:
lim(h2, x=1/9)

1/3

In [64]:
lim(h2, x=1/3)

x |--> 1

In [66]:
lim(h2, x=1/3, dir= 'right')

x |--> 1

In [67]:
lim(h2, x=1/3, dir= 'left')

x |--> 1

In [76]:
f(x)=(x^2)

In [77]:
lim(f, x=1, dir='left')

x |--> 1

In [79]:
lim(f, x=1, dir='right')

x |--> 1

In [81]:
h3(x)=2*cos(x)

In [82]:
lim(h3, x=(pi/3), dir='left')

x |--> 1

In [83]:
lim(h3, x=(pi/3), dir='right')

x |--> 1

In [84]:
h4(x)=1/x

In [85]:
lim(h4, x=0, dir='right')

x |--> +Infinity

In [86]:
lim(h4, x=0, dir='left')

x |--> -Infinity

In [89]:
lim(h4, x=0)

x |--> Infinity

In [90]:
+oo

+Infinity

In [91]:
-oo

-Infinity

In [92]:
f(x)=e^(1/x)

In [93]:
lim(f, x=0, dir='left')

x |--> 0

In [94]:
lim(f, x=0, dir='right')

x |--> +Infinity

In [95]:
lim(f, x=0)

x |--> und

In [97]:
var('p')

p

In [100]:
assume(p>1)

In [101]:
integrate(1/x**p,x,1,oo)

1/(p - 1)

In [103]:
p:forget()

In [104]:
solve(3*x+7>15,x)

[[x > (8/3)]]

In [107]:
solve(x^2+x-6==0,x)

[x == -3, x == 2]

In [108]:
solve(x^2+x-6>0,x)

[[x < -3], [x > 2]]

In [110]:
solve(x^2-2*x+1/2<0,x)

[[x > -1/2*sqrt(2) + 1, x < 1/2*sqrt(2) + 1]]

In [119]:
find_root(sin(x)==x,-1,1)

-1.0

In [149]:
solve(e^x==x,x)

[x == e^x]

In [137]:
find_root(sin(x)==cos(x),-1,1)

0.7390851332151607

In [151]:
var('y')

y

In [152]:
solve([3*x-y==2, -2*x-y==1], x,y)

[[x == (1/5), y == (-7/5)]]

In [154]:
solve([x+3==7, 9*x-y==1], x,y)

[[x == 4, y == 35]]

In [168]:
assume(x>1)
solve(x**2+3*x-5==0, x)

[x == 1/2*sqrt(29) - 3/2]

In [169]:
x:forget()

In [170]:
assume([-2<x,x<4])

In [174]:
solve((x*2)-(8*x)+1==0,x)

[x == (1/6)]

In [175]:
var('p')

p

In [179]:
solve(p*x*2+2*x+p-2==0,x)

[x == -1/2*(p - 2)/(p + 1)]

Dla jakich wartości parametru p
, równanie px2+2x+p−2=0
 ma dwa różne rozwiązania z przedziału (−∞,1).
 
 Rozwiązanie:
 Aby równanie miało 2 różne rozwiązania, to delta musi być większa od 0. Zatem:

In [210]:
assume([x<1])

In [203]:
solve(((2^2)-4*p*(p-2))>0, p)

[[p > -sqrt(2) + 1, p < sqrt(2) + 1]]

In [209]:
solve(p*x*2+2*x+p-2==0,p)

[p == -2*(x - 1)/(2*x + 1)]